tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/transforms/legalize_to_linalg.cc

```diff
@@ -1729,6 +1761,27 @@ struct NormalConvOpConversion : public OpConversionPattern<mhlo::ConvOp> {
     // TODO(ataei): Only support dilated kernel right now. We need to consider
     // input dilation for deconvolution cases.
     Attribute dilations = op.rhs_dilationAttr();
+
+    // Check if padding is zero.
+    DenseIntElementsAttr padding = op.paddingAttr();
+    if (padding && !isSplatValue(*op.padding(), 0)) {
+      if (rank != 4)
+        return rewriter.notifyMatchFailure(op, "padding expected 2d input.");
+      SmallVector<int64_t, 4> pad(2, 0);
+
+      // Store the padding values in a vector.
+      getValuesFromIntAttribute(padding, pad);
+      pad.resize(pad.size() + 2, 0);
+      // Pad the given input.
+      input = applyPad(loc, input, pad, zero_attr, rewriter);
+      // Create conv op using the padded input.
+      auto convOp = rewriter.create<linalg::Conv2DNhwcHwcfOp>(
+          loc, result_type, ValueRange{input, filter}, ValueRange{zero_tensor},
+          strides, dilations, PruneAttributeList(op));
+      rewriter.replaceOp(op, convOp.getOperation()->getResults());
+      return success();
+    }
```

Comment:
Why the special case for rank 4 only?

Reply:
Sure. We can extend this for rank 3 and rank 4 too